### PR TITLE
Add passes that add include ops

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

File: lib/Dialect/EmitC/Transforms/InsertIncludes.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+struct InsertEmitCMHLOIncludePass
+    : public InsertEmitCMHLOIncludeBase<InsertEmitCMHLOIncludePass> {
+  void runOnOperation() override {
+    auto op = getOperation();
+
+    OpBuilder builder(op);
+
+    StringRef includeName = "emitc_mhlo.h";
+    builder.setInsertionPointToStart(&op.getRegion().getBlocks().front());
+    builder.create<emitc::IncludeOp>(op.getLoc(), includeName, false);
```
[the remark this answers is not in the document]
Comment (marbre) [reply lib/Dialect/EmitC/Transforms/InsertIncludes.cpp]: Good point :)

File: include/emitc/Dialect/EmitC/Conversion/Passes.h [hunk 2]
```diff
@@ -24,6 +24,7 @@ std::unique_ptr<FunctionPass> createConvertMhloToEmitCPass();
 std::unique_ptr<FunctionPass> createConvertStdToEmitCPass();
 std::unique_ptr<FunctionPass> createConvertTensorToEmitCPass();
 std::unique_ptr<FunctionPass> createConvertTosaToEmitCPass();
+std::unique_ptr<OperationPass<ModuleOp>> createInsertEmitCTosaIncludePass();
```
[the remark this answers is not in the document]
Comment (marbre) [reply include/emitc/Dialect/EmitC/Conversion/Passes.h]: Excellent catch. Missed to delete this while refactoring.

File: include/emitc/Dialect/EmitC/Transforms/Passes.td [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef EMITC_DIALECT_EMITC_TRANSFORMS_PASSES
+#define EMITC_DIALECT_EMITC_TRANSFORMS_PASSES
+
+include "mlir/Pass/PassBase.td"
+
+def InsertEmitCMHLOInclude : Pass<"insert-emitc-mhlo-include", "ModuleOp"> {
+  let summary = "Insert an EmitC include for MHLO.";
```
Comment (simon-camp) [inline include/emitc/Dialect/EmitC/Transforms/Passes.td:15]: Maybe change this to
```
  let summary = "Insert an EmitC include for MHLO operations.";
```
or
```
  let summary = "Insert an EmitC include for the MHLO dialect.";
```

**Review by simon-camp — DISMISSED** [review]
One minor comment, feel free to ignore.

**Pull request merged**